src/scripts/core/models/grid.ts

```diff
@@ -65,10 +65,57 @@ export class Grid extends Container {
 
     }
 
+    private matchCheck(inputCoords: number[]): void {
+
+        const originX = inputCoords[0];
+        const originY = inputCoords[1];
+        const origin = this.getToken(originX, originY);
+        let xMatches = []; xMatches.push(origin);
```

Comment:
Can the .push statements be on separate lines to the declarations please?

Also you can declare these arrays using const rather than let- pushing to them doesn't count as a change (I forget this constantly).